services/submission/src/main/java/app/coronawarn/server/services/submission/verification/TanVerifier.java

```diff
@@ -19,57 +19,97 @@
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
 
 /**
  * The TanVerifier performs the verification of submission TANs.
  */
-@Component
+@Service
 public class TanVerifier {
 
   private static final Logger logger = LoggerFactory.getLogger(TanVerifier.class);
+  private final String verificationServiceUrl;
+  private final RestTemplate restTemplate;
+  private final HttpHeaders requestHeader = new HttpHeaders();
+
+  /**
+   * This class can be used to verify a TAN against a configured verification service.
+   *
+   * @param submissionServiceConfig A submission service configuration
+   * @param restTemplateBuilder A rest template builder
+   */
+  @Autowired
+  public TanVerifier(SubmissionServiceConfig submissionServiceConfig, RestTemplateBuilder restTemplateBuilder) {
+    this.verificationServiceUrl = submissionServiceConfig.getVerificationBaseUrl()
+      + submissionServiceConfig.getVerificationPath();
+    this.restTemplate = restTemplateBuilder.build();
+    this.requestHeader.setContentType(MediaType.APPLICATION_JSON);
+  }
 
   /**
    * Verifies the specified TAN. Returns {@literal true} if the specified TAN is valid, {@literal false} otherwise.
    *
-   * @param tan Submission Authorization TAN, in format "(TAN|TELETAN) KEY"
+   * @param tan Submission Authorization TAN
    * @return {@literal true} if the specified TAN is valid, {@literal false} otherwise.
+   * @throws RestClientException if status code is neither 2xx nor 4xx
    */
   public boolean verifyTan(String tan) {
-    try {
-      var tanAuthorization = TanAuthorization.of(tan);
+    String trimmedTan = tan.trim();
 
-      return validateAuthorization(tanAuthorization);
-    } catch (TanAuthorizationException e) {
-      logger.error("Failed TAN validation", e);
+    if (!checkTanSyntax(trimmedTan)) {
+      logger.debug("TAN Syntax check failed for TAN: {}", trimmedTan);
       return false;
     }
-  }
 
-  private boolean validateAuthorization(TanAuthorization tanAuthorization) {
-    var authType = tanAuthorization.getAuthType();
+    return verifyWithVerificationService(trimmedTan);
+  }
 
-    switch (authType) {
-      case TAN:
-        return validateTan(tanAuthorization.getKey());
-      case TELETAN:
-        return validateTeleTan(tanAuthorization.getKey());
-      default:
-        throw new UnsupportedOperationException("Authorization Type not supported: " + authType);
+  /**
+   * Verifies if the provided TAN can be parsed as a UUID.
+   *
+   * @param tan Submission Authorization TAN
+   * @return {@literal true} if tan can be parsed as a UUID, {@literal false} otherwise
+   */
+  private boolean checkTanSyntax(String tan) {
```

Comment:
We could consider tackling this with a ``@pattern`` annotation on the controller method.
see ``@RequestHeader(value = "cwa-authorization") @Pattern(regexp = MY_FAVORITE_URI_REGEX) String tan``

Reply:
you mean a meticulously hand-crafted regex